### PR TITLE
(DOCSP-20137): RQL Operator Expansion

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

- Document previously undocumented RQL operators
- Clarify collection operators and subqueries
- Update page formatting

### Jira

- https://jira.mongodb.org/browse/DOCSP-20137

### Staged Changes (Requires MongoDB Corp SSO)

- [Realm Query Language](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-20137/realm-query-language)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/realm-query-language.txt [hunk 1]
```diff
@@ -16,6 +14,8 @@ Realm Query Language
 Realm Query Language (RQL) is a string-based query language to constrain
 searches when retrieving objects from a realm. SDK-specific methods pass queries
 to the Realm query engine, which retrieves matching objects from the realm.
+Realm Query Language syntax is based on `NSPredicate
+<https://developer.apple.com/documentation/foundation/nspredicate>`__.
```
Comment (nlarew) [inline source/realm-query-language.txt:18]: Would be nice to have a sentence/section that outlines what's different between NSP and RQL

**Review by nlarew — APPROVED** [review]
LGTM

File: source/realm-query-language.txt (hunk 1 above)
Comment (mongodben) [reply source/realm-query-language.txt:18]: good point. and ya know what, i have no idea what that difference would be. will ask the realm sdk folks.

Comment (mongodben) [reply source/realm-query-language.txt:18]: let's split that into separate unit of work.

**Pull request merged**